src/redux/thunks/currency.ts

```diff
@@ -0,0 +1,62 @@
+import { isAxiosError } from 'axios';
+import type { ThunkType, ObjectType } from '../../utils/types/common';
+import type { CurrencyAction } from '../types/currency';
+import { CurrencyId, FieldName } from '../../utils/types/enums';
+import { getCurrencyAPI } from '../../services/api/currency/currency';
+import {
+  setCurrencyRequest,
+  setCurrencySuccess,
+  setCurrencyFailure,
+  setCurrencyOnBelarusCoin,
+  setCurrencyOnAmericaCoin,
+  setCurrencyOnEuropeCoin,
+  setCurrencyOnRussiaCoin,
+  setCurrencyOnUkraineCoin,
+  setCurrencyOnPolandCoin,
+} from '../actions/currency';
+
+export const getCurrency = (coin: string, name: string): ThunkType<CurrencyAction> => {
+  return async (dispatch) => {
+    try {
+      dispatch(setCurrencyRequest());
+      const americaCoin: number = await getCurrencyAPI(CurrencyId.usdId);
+      const europeCoin: number = await getCurrencyAPI(CurrencyId.eurId);
+      const russiaCoin: number = await getCurrencyAPI(CurrencyId.rubId);
+      const ukraineCoin: number = await getCurrencyAPI(CurrencyId.uahId);
+      const polandCoin: number = await getCurrencyAPI(CurrencyId.plnId);
```

Comment:
Can we use Promise.all here? So we could get the requests in parallel

Reply:
Done!